spec/realtime/message.test.js

```diff
@@ -70,6 +70,68 @@ define(['ably', 'shared_helper'], function(Ably, helper) {
 		}
 	};
 
+	/*
+	 * Test that a message is not sent back to the same realtime client
+	 * when echoMessages is false (RTC1a and RTL7f)
+	 *
+	 * Test that a message is sent back to the same realtime client
+	 * when echoMessages is true (RTC1a and RTL7f)
+	 */
+	exports.publishEcho = function(test) {
+		test.expect(5);
+
+		// set up two realtimes
+		var realtimeNoEcho = helper.AblyRealtime({ echoMessages: false }),
+			realtimeEcho = helper.AblyRealtime({ echoMessages: true }),
+
+			realtimeNoEchoChannel = realtimeNoEcho.channels.get('publishecho'),
+			realtimeEchoChannel = realtimeEcho.channels.get('publishecho'),
+
+			testMsg1 = 'Hello',
+			testMsg2 = 'World!';
+
+		monitorConnection(test, realtimeNoEcho, realtimeEcho);
+
+		// We expect to see testMsg2 on realtimeNoEcho and testMsg1 on both realtimeNoEcho and realtimeEcho
+		var receivedMessagesNoEcho = [],
+			receivedMessagesEcho = [];
+
+		var finishTest = function() {
+			if (receivedMessagesNoEcho.length + receivedMessagesEcho.length == 3) {
+				test.equal(receivedMessagesNoEcho.length, 1, 'Received exactly one message on realtimeNoEcho');
+				test.equal(receivedMessagesEcho.length, 2, 'Received exactly two messages on realtimeEcho');
+				try {
+					test.equal(receivedMessagesNoEcho[0].data, testMsg2, 'Received testMsg2 on realtimeNoEcho');
+					test.equal(receivedMessagesEcho[0].data, testMsg1, 'Received testMsg1 on realtimeEcho first');
+					test.equal(receivedMessagesEcho[1].data, testMsg2, 'Received testMsg2 on realtimeEcho second');
+				} catch(e) {
+					test.ok(false, 'Failed to find an expected message in the received messages');
+				}
+				closeAndFinish(test, [realtimeNoEcho, realtimeEcho]);
+			}
+		}
+
+		// first subscribe to realtimeNoEcho's channel event0
+		realtimeNoEchoChannel.subscribe('event0', function(msg) {
+			receivedMessagesNoEcho.push(msg);
+			finishTest();
+		}, function() {
```

Comment:
But that is a different point entirely @SimonWoolf.  This is a second callback on a `subscribe` method which has in implicit attach, whereas that discussion was in relation to whether we throw an exception immediately or pass the exception to the callback. This callback is unrelated to that discussion as far as I can tell?